### PR TITLE
feature(#687): Implemented gblinear model 

### PR DESCRIPTION
**Pull request by lschilders**
This pull request introduces a new linear forecaster based on XGBoost's `gblinear` booster, enhances the robustness of model serialization/deserialization for both linear and tree-based XGBoost forecasters, and adds comprehensive unit tests for the new model. It also improves the handling of sample weights in datasets and updates relevant documentation and fixtures.

**New GBLinear Forecaster and Testing:**

* Added a new `GBLinearForecaster` class implementing a gradient-boosted linear model using XGBoost's `gblinear` booster, with support for quantile regression, sample weights, and robust state serialization/deserialization.
* Introduced comprehensive unit tests for `GBLinearForecaster`, including fit/predict, state round-tripping, error handling, and sample weight effects.
* Added pytest fixtures for reusable sample datasets, including datasets with sample weights, to support new and existing forecaster tests.

**Improvements to Model State Handling and Validation:**

* Enhanced state loading for both `GBLinearForecaster` and `XGBoostForecaster` to validate the booster type when restoring from serialized state, raising a `ModelLoadingError` if the booster type does not match the expected value. [[1]](diffhunk://#diff-1725b6c0037fdfb4a0cffae2fcdfa37f98516045a969b09bf8f29b909f2fd9a5R1-R269) [[2]](diffhunk://#diff-a933e569b6f94894ab18b8d097fdf9804d98dbfb372019d35ee3781624c165a3R346-R352)
* Simplified state serialization in `XGBoostForecaster` by removing unnecessary custom attributes during model saving.

**Sample Weight Handling and Documentation:**

* Updated `ForecastInputDataset` documentation and doctests to demonstrate and validate sample weight support, including new examples for `sample_weight_column` usage.

**Documentation and Cross-Referencing:**

* Improved docstrings and cross-references between `XGBoostForecaster` and `GBLinearForecaster` for better discoverability and clarity in the codebase.

**Test Maintenance:**

* Updated imports and error handling in `test_xgboost_forecaster.py` to align with new exception handling and fixture usage.

### CODE REVIEW

File: packages/openstef-models/src/openstef_models/models/forecasting/gblinear_forecaster.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+        ...     horizons=[LeadTime(timedelta(hours=1))],
+        ...     hyperparams=GBLinearHyperParams(
+        ...         learning_rate=0.1,
+        ...         reg_alpha=0.1,
+        ...         reg_lambda=1.0
+        ...     )
+        ... )
+        >>> forecaster = GBLinearForecaster(config)
+        >>> # forecaster.fit(training_data)
+        >>> # predictions = forecaster.predict(test_data)
```
Comment (egordm) [inline packages/openstef-models/src/openstef_models/models/forecasting/gblinear_forecaster.py]: Rather than commenting these I would use`# doctest: +SKIP` after each line. This way the code still gets highlighted in the docs website.

**Review by egordm — APPROVED** [review]
Looks good! Only one nitpick. Approved in advance.
